src/sentry/snuba/metrics/naming_layer/public.py

```diff
@@ -101,6 +100,11 @@ class TransactionMetricKey(Enum):
     SPAN_USER = "span.user"
     SPAN_DURATION = "span.duration"
 
+    COUNT_ON_DEMAND = "count.on_demand"
+    DIST_ON_DEMAND = "dist.on_demand"
+    SET_ON_DEMAND = "set.on_demand"
+    GAUGE_ON_DEMAND = "gauge.on_demand"
```

Comment:
Which of the changes in the naming layer are actually required for this PR? If we'd like to expose the on-demand metrics to the metrics API, we can do so in a separate PR and fix MRI translation along the way.

Reply:
Unfortunately, `get_snql_query()` fails without this since we use `MetricsQuery` and pass it a `MetricField` whose __init__ contains [this check](https://github.com/getsentry/sentry/blob/e49a2c93bbdd2484b496a3128a311189b0e338c0/src/sentry/snuba/metrics/query.py#L48) so we can either leave this or feature-flag the check
I did remove gauge mapping in [9213141] (https://github.com/getsentry/sentry/pull/52000/commits/9213141449a0f32e9bf018505a2f206b55fbabeb)